server/protocol/protocol.py

```diff
@@ -4,10 +4,26 @@
 from asyncio import StreamReader, StreamWriter
 
 import server.metrics as metrics
+from server.config import config
 
 from ..asyncio_extensions import synchronizedmethod
 
-json_encoder = json.JSONEncoder(separators=(",", ":"))
+
+class CustomJSONEncoder(json.JSONEncoder):
+    # taken from https://stackoverflow.com/a/53798633
+    def encode(self, o):
```

Comment:
Oh sorry, I see this is the 'round floats' approach you mentioned in your other comment.

Reply:
I don't quite understand what do you mean by a wrapper, because if it requires us to call this wrapper every time we do some math, then why don't just use `round`?
Rewriting representation of the float won't help, the C code will operate with value and encode all the digits

Reply:
Something similar to this, but using our own class instead of Decimal: https://stackoverflow.com/a/3885198

```python3
class PFloat:
    def __init__(self, value: float, precision: int):
        self.value = value
        self.precision = precision
```

And then in the `to_dict` method you'd have to wrap the floats in this class:

```python3
def to_dict(self):
    return {
        "some_float": PFloat(self.some_float, 2)
    }
```

Reply:
Yeah, I can make a class, but I think it is unnecessary complication, because we already have built-in `round` function which effectively does the same thing

Reply:
Like, we have 2 options:
  1. Rewrite json's `encode` method to change encoding of all floats
  2. Rewrite every `to_dict` method where we will round every float with its own precision (the "more control" mentioned above)